source/host_resolver.c

```diff
@@ -80,6 +82,28 @@ int aws_host_resolver_purge_cache(struct aws_host_resolver *resolver) {
     return resolver->vtable->purge_cache(resolver);
 }
 
+int aws_host_resolver_purge_host_cache(
+    struct aws_host_resolver *resolver,
+    const struct aws_host_resolver_purge_host_options *options) {
+    if (!resolver) {
```

Comment:
Let's follow `aws_host_resolver_add_host_listener`. Use `AWS_PRECONDITION` to NULL check on resolver, and resolver vtable, and if there no implementation of `purge_host_cache`, raise error.

Validate the options from the implementation.